github-pullrequest-plugin/src/main/java/org/jenkinsci/plugins/github/pullrequest/trigger/check/PullRequestToCauseConverter.java

```diff
@@ -44,10 +41,15 @@ public static PullRequestToCauseConverter toGitHubPRCause(GitHubPRRepository loc
 
     @Override
     public GitHubPRCause apply(final GHPullRequest remotePR) {
-        return from(trigger.getEvents())
-                .transform(toCause(remotePR))
-                .filter(notNull())
-                .firstMatch(new SkippedCauseFilter(listener)).orNull();
+        for (GitHubPREvent event: trigger.getEvents()) {
```

Comment:
My intention was to keep the basic logic the same: return a cause if the triggers found a positive reason to approve it or return `null` if the triggers disqualify the PR. So it should be returning the same thing that it did before except that it does not evaluate trigger events that it doesn't have to -- as the documentation suggested was the case... for performance reasons :smile:

But like I said, take it or leave it. No offense taken if you want to close this PR and move ahead with #58 and/or #59.